File: Das.net.persistencia/Das.Persistencia/Persistence/ThePersistant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Das.Persistencia
{
    public class ThePersistant : IPersistence
    {
        public ThePersistant(Serializer engine)
        {
            this.PersistenceEngine = engine;
        }
        public ThePersistant()
        {
            this.PersistenceEngine = new ObjSerializerToTXT();
        }

        private Serializer PersistenceEngine
        {
            get;
            set;
        }

        public bool Save<T>(T value)
        {
            throw new NotImplementedException();
        }

        public T Get<T>(string key)
        {
            throw new NotImplementedException();
        }

        public bool Delete<T>(T value)
        {
            throw new NotImplementedException();
        }
    }
}

File: Das.java.persistencia/Das.persistencia/src/ThePersistant.java
import java.util.ArrayList;

	public class ThePersistant extends IPersistence {

		private Serializer PersistenceEngine;

		public ThePersistant() {
			this.PersistenceEngine = new ObjSerializertoXML();
		}
		
		public ThePersistant(Serializer engine){
			this.PersistenceEngine=engine;
		}

		@Override
		public <T> boolean Delete(T key) {
			return PersistenceEngine.Delete(key);
		}

		@Override
		public <T> T Get(String className, String key) {
			return PersistenceEngine.Get(className,key);
		}
		
		public <T> ArrayList<T> GetAll(String className) {
			return PersistenceEngine.GetAll(className);
		}

		@Override
		public <T> boolean Save(T value) {
			return PersistenceEngine.Save(value);
		}
	}
